src/twitch/commands.rs

```diff
@@ -107,6 +115,19 @@ impl TwitchCommand {
             _ => bail!("Invalid slow command arguments"),
         }
     }
+    fn handle_commercial_command(args: &[&str]) -> Result<Self, Error> {
+        debug!("Commercial command received as {:?}", args);
+        match args.iter().as_slice() {
+            [commercial_duration] => {
+                let duration = commercial_duration.parse::<usize>()?;
+
+                Ok(Self::Commercial(duration))
+            }
+            //TODO uh does it make sense to structure it here
```

Comment:
you could set the match statement return value to a variable, and then set `Self::Commercial` to whatever that variable is.